[navigation_links]

# Exceptions, Rescued and Not

## Example Test

[file_source](../tests/exceptions_test.rb)

Notes:

- Section `Exception rescued` rescues its exception.  The test exits the _section_, and any following code in the _section_ is not reached.
- Section 'Exception not rescued` does not rescue its exception.  The test exits entirely, and any following code in the _test_ is not reached.
- Any section, including a nested section, may rescue an exception.  (Or not -- it's an independent choice for each section.)

## Log

[file_source](../logs/test_exceptions.xml)

Notes:

- Each logged exception includes its message and backtrace.
- The second exception, the one the test didn't rescue, is actually rescued by the log itself, and of course logged.

[navigation_links]
